admin_guide/node_problem_detector.adoc

```diff
@@ -0,0 +1,317 @@
+| On-demand (based on user configuration)
+| A custom plugin monitor for node-problem-detector to invoke and check various node problems with user defined check scripts.
+
+|===
+
+//https://kubernetes.io/docs/tasks/debug-application-cluster/monitor-node-health/#node-problem-detector
+The node problem detector supports file-based kernel logging. However, it is easy to extend it to xref:admin-guide-node-problem-detector-log[support other log format].
+
+[NOTE]
+====
+The node problem detector does not support log tools, such as journald or rsyslogd.
+====
+
+[[admin-guide-node-problem-detector-gce]]
+== Running the Node Problem Detector in a GCE Cluster
+
+The node problem detector runs as a cluster addon in a xref:../install_config/configuring_gce.adoc#install-config-configuring-gce[Google Compute Engine (GCE) cluster] 
+and is enabled as a link:https://github.com/kubernetes/kubernetes/tree/master/cluster/addons/node-problem-detector[Kubernetes Addon] by default.
+
+You can disable the detector by setting the `KUBE_ENABLE_NODE_PROBLEM_DETECTOR` environment variable  in the *_kubernetes/cluster/gce/config-default.sh_* file to false:
+
+[source,yaml]
+----
+KUBE_ENABLE_NODE_PROBLEM_DETECTOR=false
+----
+
+[[admin-guide-node-problem-detector-run]]
+== Running the Node Problem Detector in Other Environments
+
+To run the Node Problem Detector, in non-GCE environments, first create a xref:../dev_guide/daemonsets#dev-guide-daemonsets[daemonset] object.
+
+When xref:../dev_guide/daemonsets.adoc#dev-guide-daemonsets[creating daemonsets], the `nodeSelector` field is used to indicate the nodes on which the daemonset should deploy.
+
+. Copy the following code to a YAML file on your master:
++
+[source,yaml]
+----
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+  name: node-problem-detector-v0.1
+  namespace: kube-system
+  labels:
+    k8s-app: node-problem-detector
+    version: v0.1
+    kubernetes.io/cluster-service: "true"
+spec:
+  template:
+    metadata:
+      labels:
+        k8s-app: node-problem-detector
+        version: v0.1
+        kubernetes.io/cluster-service: "true"
+    spec:
+      nodeSelector: <1>
+        type: infra
+      hostNetwork: true
+      containers:
+      - name: node-problem-detector
+        image: k8s.gcr.io/node-problem-detector:v0.1
+        securityContext:
+          privileged: true
+        resources:
+          limits: 
+            cpu: "200m"
+            memory: "100Mi"
+          requests:
+            cpu: "20m"
+            memory: "20Mi"
+        volumeMounts:
+        - name: log 
+          mountPath: /log 
+          readOnly: true
+      volumes:
+      - name: log
+        hostPath: <2>
+          path: /var/log/
+----
++
+<1> The node selector that determines on which nodes pod replicas should be deployed
+<2> Specify the location of your cluster log directory.
+
```

Comment:
@joelsmith Must this file be called `node-problem-detector.yaml` or can it be any name?